types/next/document.d.ts

```diff
@@ -1,15 +1,46 @@
 import * as React from "react";
+import * as http from "http";
+
+export interface Context {
+    err?: Error;
+    req: http.IncomingMessage;
+    res: http.ServerResponse;
+    pathname: string;
+    query?: {
+        [key: string]:
+            | boolean
+            | boolean[]
+            | number
+            | number[]
+            | string
+            | string[];
+    };
+    asPath: string;
+
+    renderPage(
+        enhancer?: (page: React.Component) => React.ComponentType<any>
```

Comment:
`page` should be `ComponentType<any>` instead of `Component` otherwise you get "JSX element type '…' does not have any construct or call signatures". In other words, as it is now, page is an _instance of_ Component which is not a valid JSX element.

Reply:
```

Reply:
```js
    const sheet = new ServerStyleSheet();
    const page = renderPage(App => (props: {}) =>
      sheet.collectStyles(<App {...props} />)
      //                   ^^^ I get the error here
    );
```
ServerStyleSheet comes from styled-components which I've adapted from [this example](https://github.com/zeit/next.js/blob/canary/examples/with-styled-components/pages/_document.js#L7)